data-raw/create_sdtms_data.R

```diff
@@ -0,0 +1,129 @@
+
+  # Check if the input is a valid URL
+  if (!nzchar(url) || is.null(url)) {
+    return("The source is inaccessible.")
+  } else if (!grepl(url_pattern, url)) {
+    # If the source is plain text (not a URL), return it as-is
+    return(url)
+  }
+
+  # Generate and return the hyperlink if the URL is valid
+  paste0('<a href="', url, '" target="_blank">', "Access the source of the ", link_text, " dataset.</a>")
+}
+
+
+# Main function to write documentation for a dataset
+#' @description Generate R documentation for a dataset.
+#' @param data The dataset object.
+#' @param dataset_name The name of the dataset.
+#' @param dataset_label The label for the dataset. Defaults to "No label available".
+#' @param dataset_description A description of the dataset. Defaults to "No description available".
+#' @param dataset_author The author of the dataset. Defaults to NULL if not available.
+#' @param dataset_source The source of the dataset, as a URL or text. Defaults to "No source available".
+write_doc <- function(data, dataset_name, dataset_label = "No label available",
+                      dataset_description = "No description available", dataset_author = NULL,
+                      dataset_source = "No source available") {
+  dataset_source <- generate_hyperlink(dataset_source, dataset_label)
+  doc_string <- paste(
+    "# This file is automatically generated by data-raw/create_sdtms_data.R.",
+    "# For updating it please edit inst/extdata/sdtms-specs.json and rerun create_sdtms_data.R.",
+    "# Manual edits are not recommended, as changes may be overwritten.",
+    sprintf("#' %s", dataset_label),
+    "#'",
+    sprintf("#' %s", dataset_description),
+    "#'",
+    sprintf("#' @name %s", dataset_name),
+    sprintf("#' @title %s", dataset_label),
+    sprintf("#' @keywords dataset"),
+    sprintf("#' @description dataset"),
+    "#' @docType data",
+    sprintf("#' @format A data frame with %s columns:", ncol(data)),
+    "#'   \\describe{",
+    paste(sapply(names(data), function(col_name) {
+      paste(sprintf("#'     \\item{%s}{%s}", col_name, get_attr(data, col_name)))
+    }, USE.NAMES = FALSE), collapse = "\n"),
+    "#'   }",
+    "#'",
+    sprintf("#' @source %s", dataset_source),
+    sep = "\n"
+  )
+
+  if (!is.null(dataset_author) && dataset_author != "") {
+    doc_string <- paste(doc_string, sprintf("#' @author %s", dataset_author), sep = "\n")
+  }
+  doc_string <- paste(doc_string, sprintf("\"%s\"", dataset_name), sep = "\n")
+  writeLines(doc_string, con = file.path("R", paste0(dataset_name, ".R")))
+}
+
+# List datasets and generate documentation
+datasets <- data(package = "pharmaversesdtm")$results[, "Item"]
+for (dataset_name in datasets) {
+  data(list = dataset_name, package = "pharmaversesdtm")
+  dataset <- get(dataset_name)
+  metadata <- specs[specs$name == dataset_name, ]
+
+  if (nrow(metadata) == 0) {
+    cat("No metadata found for", dataset_name, "- using default values.\n")
```

Comment:
Should this be a warning or an error?

The documentation of `rs_onco_ca125` and `supprs_onco_ca125` was removed but it was not noticed.

I've re-added it.